### PR TITLE
Reminder notifications for outstanding references

### PR DESCRIPTION
**Pull request by steventux**
[Trello](https://trello.com/c/xTeS3ggL/1593-reminder-notifications-to-outstanding-references)

## Context

Referees have 6 weeks to provide references.
Reminders sent at frequency: 2 weeks and 4 weeks

## Changes

- Makes `ReminderEmail` use a polymorphic association to requestables (ie. `FurtherInformationRequest` and `ReferenceRequest`)
- Adds service class and job to send reference request reminders
- Schedules triggering of reference request reminder job

### CODE REVIEW

File: db/migrate/20230227152913_change_reminder_emails_to_polymorphic_association.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  7.0
+]
+  def change
+    change_table :reminder_emails, bulk: true do |t|
+      t.remove_foreign_key :further_information_requests
+      t.remove_index [:further_information_request_id]
+      t.rename :further_information_request_id, :requestable_id
+      t.string :requestable_type,
+               null: false,
+               default: "FurtherInformationRequest"
```
Comment (thomasleese) [inline db/migrate/20230227152913_change_reminder_emails_to_polymorphic_association.rb:11]: Does the migration work if we then immediately after this line change the default to `nil`? I don't think it matters much, but feels a bit odd to leave the default here I think.

**Review by thomasleese — COMMENTED** [review]
One small comment, otherwise looks good to me!

File: db/migrate/20230227152913_change_reminder_emails_to_polymorphic_association.rb (hunk 1 above)
Comment (steventux) [reply db/migrate/20230227152913_change_reminder_emails_to_polymorphic_association.rb:11]: Good idea, I'll give it a go.

**Review by thomasleese — APPROVED** [review]

File: db/migrate/20230227152913_change_reminder_emails_to_polymorphic_association.rb (hunk 1 above)
Comment (steventux) [reply db/migrate/20230227152913_change_reminder_emails_to_polymorphic_association.rb:11]: It works if the `change_column_default` method is called outside of the `change_table` block ie. we issue 2 `ALTER` statements to ensure the default is applied to existing records then reset to an empty string. I've amended the commit.

**Pull request merged**